### PR TITLE
docs - more avro compression codecs, more greenplum type certs

### PR DESCRIPTION
**Pull request by lisakowen**
docs for https://github.com/greenplum-db/pxf/pull/392 (compression support) and https://github.com/greenplum-db/pxf/pull/401 (more greenplum type to avro string certification)

in this PR:
- avro supports bzip2, gz
- more greenplum types certified

### CODE REVIEW

File: docs/content/hdfs_avro.html.md.erb [hunk 1]
```diff
@@ -75,14 +75,18 @@ PXF uses the following data type mapping when writing Avro data:
...
 | real | float |
-| smallint<sup>1</sup> | int |
+| smallint<sup>2</sup> | int |
 | text | string |
+| varchar | string |
 | array ([]), enum, record | string |
+| date, time, timestamp, timestamptz | string |
 
-</br><sup>1</sup>&nbsp;PXF converts Greenplum <code>smallint</code> types to <code>int</code> before it writes the Avro data. Be sure to read the field into an <code>int</code>.
+</br><sup>1</sup>&nbsp;PXF right-pads <code>char[<i>n</i>]</code> types to length <code><i>n</i> + 1</code>, if required, with white space.
```
Comment (oliverralbertini) [inline docs/content/hdfs_avro.html.md.erb:88]: Does it really pad to n+1? I thought it would be n.

File: docs/content/hdfs_avro.html.md.erb [hunk 2]
```diff
@@ -150,7 +154,7 @@ The PXF `hdfs:avro` profile supports encoding- and compression-related write opt
 
 | Write Option  | Value Description |
 |-------|-------------------------------------|
-| COMPRESSION_CODEC    | The compression codec alias. Supported compression codecs for writing Avro data include: `snappy`, `deflate`, and `uncompressed` . If this option is not provided, PXF compresses the data using `deflate` compression. |
+| COMPRESSION_CODEC    | The compression codec alias. Supported compression codecs for writing Avro data include: `bzip2`, `xz`, `snappy`, `deflate`, and `uncompressed` . If this option is not provided, PXF compresses the data using `deflate` compression. |
 | CODEC_LEVEL    | The compression level (applicable to the `deflate` codec only). This level controls the trade-off between speed and compression. Valid values are 1 (fastest) to 9 (most compressed). The default compression level when using the `deflate` codec is 6. |
```
Comment (oliverralbertini) [inline docs/content/hdfs_avro.html.md.erb:158]: xz also would be impacted by CODEC_LEVEL

Comment (oliverralbertini) [reply docs/content/hdfs_avro.html.md.erb:158]: see here for details: https://github.com/greenplum-db/pxf/blob/e18ee9ef9ac58a1150412315befaab9c62c08bc0/server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/AvroFileAccessor.java#L162

File: docs/content/hdfs_avro.html.md.erb (hunk 1 above)
Comment (lisakowen) [reply docs/content/hdfs_avro.html.md.erb]: i saw on https://gpdb.docs.pivotal.io/6-9/ref_guide/data_types.html that the length of a char array is size of array plus one byte, assumed that would carry over.  i changed this to n.

**Comment by lisakowen** [comment]
thanks for reviewing, @oliverralbertini !

**Pull request merged**